[Messenger] Fix default serializer not handling DateTime objects properly

barton-webwings opened the pull request:
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | --
| License       | MIT
| Doc PR        | --

I was using messenger standalone and needed to send messages in JSON format. While trying to do so using the Serializer class, I noticed that DateTime objects (present for example in RedeliveryStamp) are not serialized properly.

I fixed it by adding the proper normalizer.

I encountered it in version 5.2, but it should be applicable from version 4.4.

Before:

![before](https://user-images.githubusercontent.com/109534378/192502327-22973a95-d26c-4aa7-9a12-089109fc4185.png)

After:

![after](https://user-images.githubusercontent.com/109534378/192502452-75f36ecc-a2fe-41b7-96dc-b54b0b602206.png)

barton-webwings committed:
[Messenger] Fix default serializer not handling DateTime objects properly

mtarld commented:
Hey @barton-webwings,

Thanks for your PR! As you can see in the [Symfony releases calendar](https://symfony.com/releases#symfony-releases-calendar), the 4.3 version isn't maintained anymore, can you please target 4.4 instead?

barton-webwings commented:
Hello @mtarld, 

yeah, the bot warned me about it too, so I rebased the branch onto [symfony:4.4](https://github.com/symfony/symfony/tree/4.4) and changed the target.

I am getting some weird failures in the tests though, looks like due to direct float comparison.

chalasr commented:
Don't worry about the failures, they are not related to your change.

xabbuh commented:
If you can, you could rebase your changes onto the latest `4.4` branch which should make the CI pass again.

fabpot commented:
Thank you @barton-webwings.
